tests/languages/csharp/class-name-typeof-default_feature.test

```diff
@@ -0,0 +1,61 @@
+default(int)
+default(Foo)
+typeof(Foo)
+default(Foo<Bar, IFoo<Bar>>)
+typeof(Foo<Bar, (Foo, Foo, int)>)
+
+----------------------------------------------------
+
+[
+	["keyword", "default"],
+	["punctuation", "("],
+	["keyword", "int"],
+	["punctuation", ")"],
+
+	["keyword", "default"],
+	["punctuation", "("],
+	["type-expression", ["Foo"]],
+	["punctuation", ")"],
+
+	["keyword", "typeof"],
+	["punctuation", "("],
+	["type-expression", ["Foo"]],
+	["punctuation", ")"],
+
+	["keyword", "default"],
+	["punctuation", "("],
+	["type-expression", [
+		"Foo",
+		["punctuation", "<"],
+		"Bar",
+		["punctuation", ","],
+		" IFoo",
+		["punctuation", "<"],
+		"Bar",
+		["punctuation", ">"],
+		["punctuation", ">"]
+	]],
+	["punctuation", ")"],
+
+	["keyword", "typeof"],
+	["punctuation", "("],
+	["type-expression", [
+		"Foo",
+		["punctuation", "<"],
+		"Bar",
+		["punctuation", ","],
+		["punctuation", "("],
+		"Foo",
+		["punctuation", ","],
+		" Foo",
+		["punctuation", ","],
+		["keyword", "int"],
+		["punctuation", ")"],
+		["punctuation", ">"]
+	]],
+	["punctuation", ")"]
+]
+
+----------------------------------------------------
+
+Checks for class names of type expressions.
```

Comment:
This file should be named `type-expression_feature.test` to reflect the pattern it's testing.